src/Uno.UI.Runtime.Wasm/Xaml/UIElementWasmExtensions.cs

```diff
@@ -0,0 +1,184 @@
+		}
+
+		/// <summary>
+		/// Set multiple HTML attributes to an element at the same time.
+		/// </summary>
+		public static void SetHtmlAttributes(this UIElement element, params (string name, string value)[] attributes)
+		{
+			WindowManagerInterop.SetAttributes(element.HtmlId, attributes);
+		}
+
+		/// <summary>
+		/// Clear/remove a HTML attribute from an element.
+		/// </summary>
+		public static void RemoteHtmlAttribute(this UIElement element, string name)
+		{
+			WindowManagerInterop.RemoveAttribute(element.HtmlId, name);
+		}
+
+		/// <summary>
+		/// Run javascript in the context of a DOM element.
+		/// This one is available in the scope as "element".
+		/// </summary>
+		/// <remarks>
+		/// Will work even if the element is not yet loaded into the DOM.
+		/// </remarks>
+		public static string ExecuteJavascript(this UIElement element, string jsCode)
+		{
+			var js = @$"
+(function(element) {{
+{jsCode}
+}})(Uno.UI.WindowManager.current.getView({element.HtmlId}));
+";
+			return WebAssemblyRuntime.InvokeJS(js);
+		}
+
+		/// <summary>
+		/// Asynchronously run javascript on a DOM element.
+		/// This one is available in the scope as "element".
+		/// The called code is expected to return something awaitable (a Promise).
+		/// </summary>
+		/// <remarks>
+		/// Will work even if the element is not yet loaded into the DOM.
+		/// </remarks>
+		public static Task<string> ExecuteJavascriptAsync(this UIElement element, string asyncJsCode)
+		{
+			var js = @$"
+(function(element) {{
+const __f = () => {asyncJsCode};
+return __f(element);
+}})(Uno.UI.WindowManager.current.getView({element.HtmlId}));
+";
+			return WebAssemblyRuntime.InvokeAsync(js);
+		}
+
+		/// <summary>
+		/// Set raw HTML Content for this element.
+		/// Don't use this when there's child elements managed by Uno or you'll
+		/// get expected results.
+		/// </summary>
+		public static void SetHtmlContent(this UIElement element, string html)
+		{
+			WindowManagerInterop.SetContentHtml(element.HtmlId, html);
+		}
+
+		/// <summary>
+		/// Will invoke `addEventListener(<paramref name="eventName"/>)` on the corresponding HTML element.
+		/// </summary>
+		public static void RegisterHtmlEventHandler(this UIElement element, string eventName, EventHandler handler)
+		{
+			element.RegisterEventHandler(eventName, handler);
+		}
+
+		/// <summary>
+		/// Unregister previously registered event with RegisterHtmlEventHandler.
+		/// </summary>
+		public static void UnregisterHtmlEventHandler(this UIElement element, string eventName, EventHandler handler)
+		{
+			element.UnregisterEventHandler(eventName, handler);
+		}
+
+		/// <summary>
+		/// Will invoke `addEventListener(<paramref name="eventName"/>)` on the corresponding HTML element.
+		/// </summary>
+		/// <remarks>
+		/// For use with CustomEvent("name", {detail:{detail here}}).
+		/// </remarks>
+		/// <param name="isDetailJson">
+		/// True will JSON.stringify the content. False will treat the content as a string.
```

Comment:
Could you please explain what is the "content"? I've no idea here.

Reply:
Fixed.